progress/progress.go

```diff
@@ -0,0 +1,192 @@
+package progress
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/lucasb-eyer/go-colorful"
+	"github.com/muesli/reflow/ansi"
+	"github.com/muesli/termenv"
+)
+
+const defaultWidth = 40
+
+var color func(string) termenv.Color = termenv.ColorProfile().Color
+
+// Option is used to set options in NewModel. For example:
+//
+//     progress := NewModel(
+//	       WithRamp("#ff0000", "#0000ff"),
+//	       WithoutPercentage(),
+//     )
+type Option func(*Model) error
+
+// WithDefaultRamp sets a gradient fill with default colors.
+func WithDefaultRamp() Option {
+	return WithRamp("#00dbde", "#fc00ff")
+}
+
+// WithRamp sets a gradient fill blending between two colors.
+func WithRamp(colorA, colorB string) Option {
+	return func(m *Model) error {
+		return m.setRamp(colorA, colorB, false)
+	}
+}
+
+// WithDefaultScaledRamp sets a gradient with default colors, and scales the
+// gradient to fit the filled portion of the ramp.
+func WithDefaultScaledRamp() Option {
+	return WithScaledRamp("#00dbde", "#fc00ff")
+}
+
+// WithScaledRamp scales the gradient to fit the width of the filled portion of
+// the progress bar.
+func WithScaledRamp(colorA, colorB string) Option {
```

Comment:
That's called "gradient" actually, "ramp" term can be ambiguous

Reply:
The terms are fairly interchangeable; "gradient" is more common with web designers while "ramp" is more common in motion graphics. You have a point, though: most users in this case will probably be more familiar with the term "gradient."